src/basic_recipes/tricontourf.jl

```diff
@@ -0,0 +1,198 @@
+        levels_scaled = (levels .- minimum(levels)) ./ (maximum(levels) - minimum(levels))
+        n = length(levels_scaled)
+
+        if elow == :auto && !(ehigh == :auto)
+            cm_base = cgrad(cmap, n + 1; categorical=true)[2:end]
+            cm = cgrad(cm_base, levels_scaled; categorical=true)
+        elseif ehigh == :auto && !(elow == :auto)
+            cm_base = cgrad(cmap, n + 1; categorical=true)[1:(end - 1)]
+            cm = cgrad(cm_base, levels_scaled; categorical=true)
+        elseif ehigh == :auto && elow == :auto
+            cm_base = cgrad(cmap, n + 2; categorical=true)[2:(end - 1)]
+            cm = cgrad(cm_base, levels_scaled; categorical=true)
+        else
+            cm = cgrad(cmap, levels_scaled; categorical=true)
+        end
+        return cm
+    end
+    c.attributes[:_computed_colormap] = computed_colormap
+
+    lowcolor = Observable{RGBAf}()
+    map!(lowcolor, c.extendlow, c.colormap) do el, cmap
+        if isnothing(el)
+            return RGBAf(0, 0, 0, 0)
+        elseif el === automatic || el == :auto
+            return RGBAf(to_colormap(cmap)[begin])
+        else
+            return to_color(el)::RGBAf
+        end
+    end
+    c.attributes[:_computed_extendlow] = lowcolor
+    is_extended_low = lift(x -> !isnothing(x), c.extendlow)
+
+    highcolor = Observable{RGBAf}()
+    map!(highcolor, c.extendhigh, c.colormap) do eh, cmap
+        if isnothing(eh)
+            return RGBAf(0, 0, 0, 0)
+        elseif eh === automatic || eh == :auto
+            return RGBAf(to_colormap(cmap)[end])
+        else
+            return to_color(eh)::RGBAf
+        end
+    end
+    c.attributes[:_computed_extendhigh] = highcolor
+    is_extended_high = lift(x -> !isnothing(x), c.extendhigh)
+
+    PolyType = typeof(Polygon(Point2f[], [Point2f[]]))
+
+    polys = Observable(PolyType[])
+    colors = Observable(Float64[])
+
+    function calculate_polys(xs, ys, zs, levels::Vector{Float32}, is_extended_low, is_extended_high, triangulation)
+        empty!(polys[])
+        empty!(colors[])
+
+        levels = copy(levels)
+        @assert issorted(levels)
+        is_extended_low && pushfirst!(levels, -Inf)
+        is_extended_high && push!(levels, Inf)
+        lows = levels[1:end-1]
+        highs = levels[2:end]
+
+        trianglelist = compute_triangulation(triangulation, xs, ys)
+        filledcontours = filled_tricontours(xs, ys, zs, trianglelist, levels)
+
+        levelcenters = (highs .+ lows) ./ 2
+
+        for (fc, lc) in zip(filledcontours, levelcenters)
+            pointvecs = map(fc.polylines) do vecs
+                map(vecs) do tupl
+                    Point2f(tupl...)
+                end
+            end
+            if isempty(pointvecs)
+                continue
+            end
+            
+            p = Makie.Polygon(pointvecs[1], pointvecs[2:end])
+            push!(polys[], p)
+            push!(colors[], lc)
```

Comment:
```suggestion
            
            for pointvec in pointvecs
                p = Makie.Polygon(pointvec)
                push!(polys[], p)
                push!(colors[], lc)
            end
```
seems to fix the GLMakie issue for me. Not sure if this is the correct approach in general though.